CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/screens/main/MainViewModel.kt

```diff
@@ -0,0 +1,17 @@
+package co.nimblehq.coroutine.ui.screens.main
+
+import android.content.Context
+import co.nimblehq.coroutine.R
+import co.nimblehq.coroutine.ui.base.BaseViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    @ApplicationContext val context: Context
```

Comment:
I guess it should be a warning here, HiltVM doesn't accept to keep context instance

Reply:
Hmmm, any idea why I don't see any warning? 🙏 

<img width="1440" alt="Screen Shot 2564-03-19 at 18 14 44" src="https://user-images.githubusercontent.com/18277915/111772064-04876a00-88df-11eb-8c65-ad6ae3a6d412.png">

Reply:
@Tuubz here it is https://stackoverflow.com/questions/66216839/inject-context-with-hilt-this-field-leaks-a-context-object

![image](https://user-images.githubusercontent.com/16315358/111966627-b7e39f00-8b29-11eb-9c5c-0d1365a09198.png)

Reply:
Hmmmm, trying to figure out why I don't see that same warning. Which version of Android Studio are you using?

I'm using Android Studio 4.1 🙏

Reply:
Alright, I updated the AS and I'm able to see the same warning now. 

Do you happen to have a solution for this as well? 🙈

Reply:
@Tuubz cool 🧑‍🍳  to solve this we could:
- remove context usage, back to a simple example
- use `LocalizedResources` with WeakReference of context, same to what we did in The1
- pass context into VM's fun from View layer, instead of keeping context in VM's constructor

Happy coding ;)